mongo/options/clientoptions.go

```diff
@@ -546,6 +546,11 @@ func (c *ClientOptionsBuilder) Validate() error {
 		}
 	}
 
+	if args.HeartbeatInterval != nil && *args.HeartbeatInterval < (500*time.Millisecond) {
+		return fmt.Errorf("HeartbeatFrequencyMS cannot be less than minHeartbeatFrequencyMS (500ms), got HeartbeatFrequencyMS=%q",
```

Comment:
The query parameter is [`heartbeatFrequencyMS`](https://www.mongodb.com/docs/v6.2/reference/connection-string/#mongodb-urioption-urioption.heartbeatFrequencyMS), suggest using the literal value. Also, `minHeartbeatFrequencyMS` is confusing as this is not configurable by the user and is not documented.

```suggestion
		return fmt.Errorf("heartbeatFrequencyMS must exceed the minimum heartbeat interval of 500ms, got heartbeatFrequencyMS=%q",
```

Reply:
Gotcha, also going to go ahead and rename the function and arg to clear up possible confusion!